oauth2/oauth2_auth_code_test.go

```diff
@@ -1033,12 +1034,63 @@ func TestAuthCodeWithMockStrategy(t *testing.T) {
 						hs := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 							assert.Equal(t, r.Header.Get("Content-Type"), "application/json; charset=UTF-8")
 
+							expectedGrantedScopes := []string{"openid", "offline", "hydra.*"}
+							expectedSubject := "foo"
+
 							var hookReq hydraoauth2.RefreshTokenHookRequest
 							require.NoError(t, json.NewDecoder(r.Body).Decode(&hookReq))
-							require.Equal(t, hookReq.Subject, "foo")
-							require.ElementsMatch(t, hookReq.GrantedScopes, []string{"openid", "offline", "hydra.*"})
+							require.Equal(t, hookReq.Subject, expectedSubject)
+							require.ElementsMatch(t, hookReq.GrantedScopes, expectedGrantedScopes)
 							require.ElementsMatch(t, hookReq.GrantedAudience, []string{})
 							require.Equal(t, hookReq.ClientID, oauthConfig.ClientID)
+							require.NotEmpty(t, hookReq.Session)
+							require.Equal(t, hookReq.Session.Subject, expectedSubject)
+							require.Equal(t, hookReq.Session.ClientID, oauthConfig.ClientID)
+							require.Equal(t, hookReq.Session.Extra, map[string]interface{}{})
+							require.NotEmpty(t, hookReq.Requester)
+							require.Equal(t, hookReq.Requester.ClientID, oauthConfig.ClientID)
+							require.ElementsMatch(t, hookReq.Requester.GrantedScopes, expectedGrantedScopes)
+
+							assertx.EqualAsJSONExcept(t, hookReq.Session, json.RawMessage(`{
```

Comment:
If you use `snapshotx.` it will create a JSON file on the disk which makes the content a bit easier to read:

```
	snapshotx.SnapshotT(t, json.RawMessage("..."), snapshotx.ExceptPaths("...") )
```

To update snapshots set env var `UPDATE_SNAPSHOTS=true`

Reply:
It looks like `snapshotx.SnapshotT` is not available in the version of `snapshotx` that `hydra` uses (`0.0.368`). Tried to bump to lastest - `0.0.450` - a lot of tests fail. So I used `SnapshotTExcept`.

Also, it generated 6 snapshots with the same content - probably because tests are run in parallel.